clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -247,9 +248,10 @@
     public static final String ENABLE_IDEMPOTENCE_DOC = "When set to 'true', the producer will ensure that exactly one copy of each message is written in the stream. If 'false', producer "
                                                         + "retries due to broker failures, etc., may write duplicates of the retried message in the stream. "
                                                         + "Note that enabling idempotence requires <code>" + MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION + "</code> to be less than or equal to 5, "
-                                                        + "<code>" + RETRIES_CONFIG + "</code> to be greater than 0 and <code>" + ACKS_CONFIG + "</code> must be 'all'. If these values "
+                                                        + "<code>" + RETRIES_CONFIG + "</code> to be greater than 0, and <code>" + ACKS_CONFIG + "</code> must be 'all'. If these values "
                                                         + "are not explicitly set by the user, suitable values will be chosen. If incompatible values are set, "
-                                                        + "a <code>ConfigException</code> will be thrown.";
+                                                        + "a <code>ConfigException</code> will be thrown. With an idempotent producer, setting the <code>" + MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION
+                                                        + "</code> greater than 1 will not break ordering guarantees.";
```

Comment:
nit: max-in-flight cannot be larger than 5 if idempotent producer is enabled.

Reply:
That requirement is covered a few sentences before this one, it might be cut off by Github.  I think it's clear that by "greater than 1", we just mean "between 1 and 5, the maximum allowable limit as stated above"

Reply:
Instead of an additional sentence at the end, could we modify the original sentence to also mention ordering?

Reply:
will do